### ISSUE TITLE
a11y + UX : The terms agreement checkbox should be a required field

### ISSUE THREAD

**Issue by ramiy** [WordImpress/Give#1200]
## Issue Overview

The checkbox  should have a `required` attribute.

This way, if not checked, the browser will show a warning:

![111](https://cloud.githubusercontent.com/assets/576623/19926165/c423adea-a0fb-11e6-95b4-c0bb2cc04948.png)

For better accessibility we should also use the `aria-required="true"`.

## Current Behavior
Currently I can press the "Donate Now" button without checking the "Terms" checkbox.

## Expected Behavior
The browser should alert when the required fields are empty.

## Related PRs
List related PRs against other branches: #1178  #1193

**Comment by mathetos** [comment]
@ramiy It just needs to conform to the new format for HTML5 alerts, per this issue:
https://github.com/WordImpress/Give/issues/351

**Comment by ramiy** [comment]
@mathetos I've seen the HTML5 alerts, @ravinderk did a good job at #907. I've implemented the logic on the checkbox.

**Comment by ramiy** [comment]
The new alert message :

![new-give-terms](https://cloud.githubusercontent.com/assets/576623/20286143/084119dc-aace-11e6-8384-5314638f1f2f.png)

The alert message moved down because of the custom zoom settings on my laptop.
